pkg/controller/controller_instance_test.go

```diff
@@ -1136,6 +1225,133 @@ func TestPollServiceInstanceSuccessDeprovisioningWithOperationWithFinalizer(t *t
 	assertNumEvents(t, events, 1)
 }
 
+func TestSetInstanceCondition(t *testing.T) {
```

Comment:
I'm cool with the use of the internal closures here, but it's a lot of state to load while reading the test. A quick summary near the case runner would help. That's where we'll see any future failures actually happen.